robyn_rate_limits/in_memory.py

```diff
@@ -1,14 +1,92 @@
+from typing import List
+from typing import NamedTuple
+from typing import Optional
+
 import cachetools
 
+from robyn_rate_limits.protocols import LimitStore
+
+
+class InMemoryStore(LimitStore):
```

Comment:
When working with protocols, you don't need to inherit from the base class, adhering to the protocol is implied by shape.
This is true for all of the rest of the implementations, I will not litter the review with this comment, but it's true to all of them.